### PR TITLE
Updates, Review System fix and rework.

### PR DESCRIPTION
**Pull request by Syreldar**
**Main:**
- Moved code into a static method "MainMenu" inside Utilities.

**Chart:**
- Removed addReview().
- Added addShopReview() and addProductReview().

**Customer**
- Removed database field and relative constructor.

**Database:**
- Removed productReviewCount, shopReviewCount fields.
- Added shopReviews, productReviews fields;
- Changed constructor accordingly.
- add* methods renamed to register*
- Updated usage of Java streams where possible.

**Product:**
- Changed reviews field and methods to take ProductReview objects.
- Added GetReviews() method.

**ProductReview:**
- Added register override.

**Review:**
- Removed register override and rendered class abstract.

**Shop:**
- Changed reviews field and methods to take ShopReview objects.
- Added GetReviews() method.

**ShopReview:**
- Added register override.

**Utilities:**
- Updated to use Java Streams where possible.
- Adapted to the rest of the code changes.
- askContinueOrExitProgram() now takes scanner as a parameter in order to not generate a new one on call.
- Reworked Review System.
- Implemented "Go back" option on Reviews menu and Charts menu.

### CODE REVIEW

File: src/main/java/org/project/Utilities.java [hunk 1]
```diff
@@ -2,22 +2,25 @@
...
+    private static void askContinueOrExitProgram(Scanner scanner)
+    {
+        System.out.println("Do you want to continue? (y/n) ");
         String answer = scanner.nextLine();
+
         if (answer.equalsIgnoreCase("n")) {
             System.out.println(GOODBYE_MESSAGE);
             System.exit(0);
         }
     }
```
Comment (Andrea2423) [inline src/main/java/org/project/Utilities.java:20]: Non è presente cosa succede se si digita la "y", il programma continua anche con una qualsiasi lettera(quando dovrebbe proseguire solo al digitare della y o Y, e se inseriamo una lettera diversa da y o n il programma dovrebbe fermarsi.

**Review by Sara-hub96 — DISMISSED** [review]
Va bene

**Review by Andrea2423 — CHANGES_REQUESTED** [review]
per me va bene, l' unica aggiungere due else dove ti ho segnato ;)

File: src/main/java/org/project/Utilities.java (hunk 1 above)
Comment (Syreldar) [reply src/main/java/org/project/Utilities.java:23]: https://github.com/Syreldar/team-project-java15/pull/18/commits/b45e855fec1dfdc51498f628d9fb4db9379328a0

Comment (Andrea2423) [reply src/main/java/org/project/Utilities.java:23]: Perfetto! ancora meglio con uno!

**Review by Andrea2423 — APPROVED** [review]
LGTM

**Pull request merged**